code/iaas/logic/src/main/java/io/cattle/platform/process/secret/SecretRemove.java

```diff
@@ -25,13 +26,15 @@
 
     @Override
     public HandlerResult handle(ProcessState state, ProcessInstance process) {
-        Secret secret = (Secret)state.getResource();
-        try {
-            secretsService.delete(secret.getAccountId(), secret.getValue());
-        } catch (IOException e) {
-            log.error("Failed to delete secret from storage [{}]",
-                    secret.getId(), e);
-            throw new IllegalStateException(e);
+        Secret secret = (Secret) state.getResource();
+        String secretValue = secret.getValue();
+        if (StringUtils.isNotBlank(secretValue)) {
```

Comment:
@leodotcloud we have to let qa know to test the scenario when empty secret was created prior to the fix, and then after performing the upgrade the secret should be removable.

Reply:
yup, will work with QA for the validation.